### PR TITLE
#162949204 Configure backend to use a Postgres database

### PR DESCRIPTION
**Pull request by Allan690**
#### What does this PR do?
Configures the backend of the application to use Postgres as it's database
#### Description of Task to be completed?
- Integrate a Postgres database and make the migrations
- add .env file with local db configurations
- Change database settings in the settings.py file
#### How should this be manually tested?
1. Clone the repo: `git clone https://github.com/andela/ah-the-unsullied.git`
2. Make cloned directory the active directory: `cd ah-the-unsullied`
3. Checkout to the current branch: `git checkout ch-configure-postgres-db-162949204`
4. Set up a virtual environment and activate it.
```
- virtualenv <name_of_env>
- source <name_of_env>/bin/activate
```
5. Create a  .env file by doing `touch .env` in the current directory(root) and enter the database connection information as shown below: 
```
export DATABASE_URL=postgres://USERNAME:PASSWORD@HOST:5432/NAME
```
6. Create the database on terminal: `psql ah_unsullied`
7.  Export the environment variables: `source .env`
8. Make migrations: `python manage.py makemigrations && python manage.py migrate`
9. Create a super user: `python manage.py createsuperuser`
10. Run the server: `python manage.py runserver`
11. Login with the credentials of the super user by navigating to: `localhost:8000/admins`
#### Any background context you want to provide?
PostgreSQL is Open Source, fully featured and optimized for production-ready environments as compared to SQLite that comes as default with Django
#### What are the relevant pivotal tracker stories?
[#162949204](https://www.pivotaltracker.com/story/show/162949204)

### CODE REVIEW

**Review by indungu — DISMISSED** [review]
LGTM

**Review by kwanj-k — CHANGES_REQUESTED** [review]
We have to get this build to pass

**Review by MbugwaSami — COMMENTED** [review]
LGTM

**Review by MbugwaSami — DISMISSED** [review]

**Review by kwanj-k — DISMISSED** [review]

**Review by nesh-dev — DISMISSED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -7,22 +7,22 @@ python:
...
- - psql -c 'create database ah_unsullied_test' -U postgres
+ - psql -c "create database ah_theunsullied;" -U postgres
 
 # Run tests
 script:
-  - python manage.py makemigrations 
-  - python manage.py migrate 
-  - coverage run --source=authors/ manage.py test 
-  - coverage report
+  - export SECRET_KEY="iamasecrettheunsadtheunsulliedtheundead"
```
Comment (indungu) [inline .travis.yml]: For security purposes, you should never have the secret-key exposed. Please remove it and have it in the environment variables

File: .travis.yml [hunk 2]
```diff
@@ -7,22 +7,22 @@ python:
...
+  - export DATABASE_URL="postgres://@localhost:5432/ah_theunsullied"
```
Comment (indungu) [inline .travis.yml]: Same applies for the DATABASE_URL, please have this added to the environment variables.

**Review by indungu — CHANGES_REQUESTED** [review]
Nice work so far, just address the following notes.

**Review by kelvinwm — DISMISSED** [review]

File: .travis.yml (hunk 2 above)
Comment (Allan690) [reply .travis.yml]: Okay...Thanks, working on it

**Review by sylviawanjiku — DISMISSED** [review]

**Comment by indungu** [comment]
Coverage report is required, please check the `.travis.yml` file and/or talk to @MbugwaSami and make the necessary changes.

**Review by indungu — APPROVED** [review]

**Review by kwanj-k — APPROVED** [review]

**Review by nesh-dev — APPROVED** [review]

**Review by sylviawanjiku — APPROVED** [review]

**Review by kelvinwm — APPROVED** [review]

**Pull request merged**